src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy

```diff
@@ -61,6 +62,42 @@ class Application {
         }
     }
 
+    List<MuleComponent> findComponents(String componentType, String namespace) {
+        List<MuleComponent> comps = []
+        configurationFiles.each { configFile ->
+            comps += configFile.findComponents(componentType, namespace)
+        }
+        return comps
```

Comment:
Would the .collect() and flatten() methods work here?  `return configurationFiles.collect { configFile.findComponents(componentType, namespace) }.flatten()`

Reply:
This pattern applies to the getFlows, getSubFlows, and getFlowrefs as well.

Reply:
Yap it does. WOW! much more simplified.